engine/src/main/java/org/apache/apex/engine/YarnAppLauncherImpl.java

```diff
@@ -164,8 +160,12 @@ public void shutdown(org.apache.apex.api.Launcher.ShutdownMode shutdownMode)
         throws org.apache.apex.api.Launcher.LauncherException
     {
       shutdownApp(this, shutdownMode);
-
     }
 
+    @Override
+    public void close() throws Exception
```

Comment:
YarnException doesn't derive from the IOException hierarchy and since this client is related to Yarn functionality, I had decided to keep the option open without having to wrap everything into an IOException in case we need to deal with those in future. That's why I chose to implement AutoCloseable instead of Closeable.